main.cpp

```diff
@@ -54,11 +54,107 @@ static const char *RcsId = "$Id$";
 #include <tango.h>
 #include <dserverclass.h>
 #include <DataBase.h>
+#include <arpa/inet.h>
+
 
 #ifndef WIN32
 #include <sys/resource.h>
 #endif
 
+#include <omniORB4/internal/giopStrand.h>
+#include <omniORB4/internal/giopStream.h>
+#include <omniORB4/internal/GIOP_S.h>
+
+#include <omniORB4/omniInterceptors.h>
+
+pthread_key_t key;
+bool is_acl_enabled;
+std::set<std::string> ip_addresses_allowed;
+
+CORBA::Boolean retrieve_client_address(omni::omniInterceptors::serverReceiveRequest_T::info_T& info)
```

Comment:
I think something like this is already implemented in the kernel. I guess that can be used in allow_command (or is_command_allowed). This will guarantee that every command is validated against the policy without inserting `check_ip` everywhere. Can be done in a separate PR though.

Reply:
We would prefer avoid that approach because we don't want check all Tangodatabaseds commands indiscriminately but only the ones which make changes to the RDBMS. In other words readings are always allowed plus a couple of commands that if not allowed make Tango stops to work.

Reply:
I don't know whether this is what Igor was suggesting but indeed the best location for the check_ip() calls would be in DataBaseStateMachine.cpp file in the is_xxx_allowed() methods (e.g.: DataBase::is_DbPutAttributeAlias_allowed ).
This would be cleaner and indeed could be done in separate PR.

Reply:
Hi @bourtemb ,
if you mean to move the check in the is_xxx_alowed method and return just true/false we would loose some information we put in the exception description (the ip of the caller that is not allowed) that could be useful.
I am not sure of the behavior if we throw an exception in the is_xxx_allowed method.

Reply:
I agree with you on the fact we should not lose the information coming from the exception.
I was not considering returning false but throwing the exception in the is_xxx_allowed method.
I tested on a standard server and the exception is correctly propagated to the client.